kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt

```diff
@@ -1331,7 +1331,7 @@ internal abstract class FunctionGenerationContext(
             generationState.dependenciesTracker.add(irClass)
             if (irClass.isObjCMetaClass()) {
                 val name = irClass.getExternalObjCMetaClassBinaryName()
-                val objCClass = getObjCClass(name)
+                val objCClass = getObjCClass(name, isMetaclass = true)
```

Comment:
This change doesn't seem right: now the following code
```kotlin
val nsObjectClass: Any = NSObject
println(nsObjectClass is NSNumberMeta)
```
prints `true`, which is not correct. It should print `false`, as it did before. Please add such a test case.

Reply:
`object_getClass` returns meta class, These pieces of code has been reverted.